calc.go

```diff
@@ -14581,6 +14440,151 @@ func (fn *formulaFuncs) UNICODE(argsList *list.List) formulaArg {
 	return fn.code("UNICODE", argsList)
 }
 
+// UNIQUE function returns a list of unique values in a list or range.
+// For syntax refer to
+// https://support.microsoft.com/en-us/office/unique-function-c5ab87fd-30a3-4ce9-9d1a-40204fb85e1e.
+func (fn *formulaFuncs) UNIQUE(argsList *list.List) formulaArg {
+	args, errArg := getFormulaUniqueArgs(argsList)
+	if errArg != nil {
+		return *errArg
+	}
+
+	if args.byColumn {
+		args.cellRange, args.cols, args.rows = transposeFormulaArgsList(args.cellRange, args.cols, args.rows)
+	}
+
+	counts := map[string]int{}
+
+	for i := 0; i < len(args.cellRange); i += args.cols {
+		key := concatValues(args.cellRange[i : i+args.cols])
+
+		if _, ok := counts[key]; !ok {
+			counts[key] = 0
+		}
+		counts[key]++
+	}
+
+	uniqueAxes := [][]formulaArg{}
+
+	for i := 0; i < len(args.cellRange); i += args.cols {
+		key := concatValues(args.cellRange[i : i+args.cols])
+
+		if (args.exactlyOnce && counts[key] == 1) || (!args.exactlyOnce && counts[key] >= 1) {
+			uniqueAxes = append(uniqueAxes, args.cellRange[i:i+args.cols])
+		}
+		delete(counts, key)
+	}
+
+	if args.byColumn {
+		uniqueAxes = transposeFormulaArgsMatrix(uniqueAxes)
+	}
+
+	return newMatrixFormulaArg(uniqueAxes)
+}
+
+func transposeFormulaArgsMatrix(args [][]formulaArg) [][]formulaArg {
+	if len(args) == 0 {
+		return args
+	}
```

Comment:
Hi @IvanHristov98, could you grant a push permission on this PR branch, I'll add unit test for these uncovered lines.